deploy.ts

```diff
@@ -44,64 +44,61 @@ let commands: ApplicationCommandsManager
  *
  * @param options Initialization options
  */
-export function init(options: { env: boolean; path?: string }): void
+export function init(options: { env: boolean; path?: string; }): void;
 export function init(options: {
-  publicKey: string
-  token?: string
-  path?: string
-}): void
+  publicKey: string;
+  token?: string;
+  path?: string;
+}): void;
 export function init(options: DeploySlashInitOptions): void {
-  if (client !== undefined) throw new Error('Already initialized')
+  if (client !== undefined) throw new Error('Already initialized');
   if (options.env === true) {
-    options.publicKey = Deno.env.get('PUBLIC_KEY')
-    options.token = Deno.env.get('TOKEN')
+    options.publicKey = Deno.env.get('PUBLIC_KEY');
+    options.token = Deno.env.get('TOKEN');
   }
 
   if (options.publicKey === undefined)
-    throw new Error('Public Key not provided')
+    throw new Error('Public Key not provided');
 
   client = new InteractionsClient({
     token: options.token,
     publicKey: options.publicKey
-  })
+  });
 
-  commands = client.commands
+  commands = client.commands;
 
-  const cb = async (evt: {
-    respondWith: CallableFunction
-    request: Request
-  }): Promise<void> => {
+  const requestHandler = async (request: Request): Promise<Response> => {
     if (options.path !== undefined) {
-      if (new URL(evt.request.url).pathname !== options.path) return
+      if (new URL(request.url).pathname !== options.path) return new Response('Bad Request', {
+        status: 400
+      });
     }
     try {
-      // we have to wrap because there are some weird scope errors
-      const d = await client.verifyFetchEvent({
-        respondWith: (...args: unknown[]) => evt.respondWith(...args),
-        request: evt.request
-      })
-      if (d === false) {
-        await evt.respondWith(
-          new Response('Not Authorized', {
-            status: 400
-          })
-        )
-        return
+      const interaction = await client.verifyFetchEvent({ request, respondWith: (res: Response) => { return res; } });
+      if (!interaction) {
```

Comment:
This here should be `interaction == false`

Reply:
You mean "interaction === false".

Reply:
Also, fixed.